pxr/usd/plugin/usdMtlx/reader.cpp

```diff
@@ -1284,7 +1299,7 @@ _Context::AddShaderNode(const mx::ConstNodePtr& mtlxShaderNode)
 
     // Get the nodeDef for this shaderNode.
     mx::ConstNodeDefPtr mtlxNodeDef = mtlxShaderNode->getNodeDef();
-    if (mtlxShaderNode->getNodeDefString().empty()) {
+    if (!mtlxNodeDef) {
```

Comment:
I wanted to double check here that this handles finding the <nodegraph>/<nodedef> associations found on <implementations>.

Reply:
Context: at this exact point in the code, the MaterialX library has not been loaded. This means that a node that has a NodeDef string will fail to resolve, and previously would not be tried again. Changing the test to retry when the NodeDef was not found allows trying a second time to find the NodeDef using the string, but this time with libraries loaded.